az-digital/az_quickstart
# Fixes #2233 Make Publication citation style not dependent on View Mode

## Description
Currently, publication citations are dependent on publication view mode. This works, but results in many extra view displays being needed to control rendering in these view modes, which is not ideal from a maintainability standpoint. It also makes it more difficult to add new citation styles to the site.

This PR adds a configuration settings form to set the site default citation style, and a special views argument `az_citation_style_argument` handler that selects citation style. This means that citation style for views can be made to vary from the site default by passing an argument to the view.

**Note: the changes in this PR are based upon the changes in #2231 and it is best reviewed compared to that branch.**

## Related issues
#2233 

## How to test

- enable `az_publication`
- create some publications
- select different citation styles on the publication settings form at `/admin/config/az-quickstart/settings/az-publication`
- verify that publications are rendered in the chosen citation style
- create a flexible page with a view reference added to `AZ Publications`. Set the view argument to the name of a citation style, e.g. `az_mla` or `az_ieee`, etc. Citation styles are listed at `/admin/structure/az_citation_style`
- verify that the view reference renders in the selected citation style.

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
